File: src/main/kotlin/org/wahlen/asyncweb/model/Product.kt
package org.wahlen.asyncweb.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table("product")
data class Product(
  @Id
  val id: Long? = null,
  val categoryId: Long,
  val name: String,
  val description: String,
  val price: Double
)

File: webapp/dtos/product.ts
export interface Product {
    id: number;
    categoryId: number;
    categoryName: string;
    name: string;
    price: number;
}

export interface ProductCreateDTO {
    categoryId: number | null;
    name: string;
    price: number | string;
}

export interface ProductUpdateDTO {
    categoryId: number | null;
    name: string;
    price: number | string;
}
